generators/import-jdl/index.js

```diff
@@ -71,80 +77,69 @@ module.exports = class extends BaseGenerator {
             },
 
             getConfig() {
-                this.applicationType = this.config.get('applicationType');
-                this.baseName = this.config.get('baseName');
-                this.databaseType = this.config.get('databaseType') || this.getDBTypeFromDBValue(this.options.db);
-                this.prodDatabaseType = this.config.get('prodDatabaseType') || this.options.db;
-                this.devDatabaseType = this.config.get('devDatabaseType') || this.options.db;
-                this.skipClient = this.config.get('skipClient');
-                this.clientFramework = this.config.get('clientFramework');
+                if (!jhiCore.FileUtils.doesFileExist('.yo-rc.json') && !this.generated) {
+                    // we may have to parse the JDL first
+                    return;
+                }
+                let configuration = this.config.getAll();
+                if (!configuration.baseName && jhiCore.FileUtils.doesFileExist('.yo-rc.json')) {
+                    configuration = JSON.parse(fs.readFileSync('.yo-rc.json', { encoding: 'utf-8' }))['generator-jhipster'];
+                    configuration.getAll = () => configuration;
```

Comment:
if such an update is needed, can we do it in a util method in generator-base and reuse it where needed instead of repeating this?

Reply:
yes